src/main/kotlin/ar/edu/unahur/obj2/vendedores/Vendedor.kt

```diff
@@ -28,24 +28,60 @@ abstract class Vendedor {
   fun otrasCertificaciones() = certificaciones.count { !it.esDeProducto }
 
   fun puntajeCertificaciones() = certificaciones.sumBy { c -> c.puntaje }
+
+  abstract fun esInfluyente():Boolean
+  fun esGenerico() = certificaciones.any { v -> !v.esDeProducto }
 }
 
 // En los parámetros, es obligatorio poner el tipo
 class VendedorFijo(val ciudadOrigen: Ciudad) : Vendedor() {
   override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
     return ciudad == ciudadOrigen
   }
+
+  override fun esInfluyente(): Boolean {
+    return false
+  }
 }
 
 // A este tipo de List no se le pueden agregar elementos una vez definida
 class Viajante(val provinciasHabilitadas: List<Provincia>) : Vendedor() {
   override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
     return provinciasHabilitadas.contains(ciudad.provincia)
   }
+  override fun esInfluyente(): Boolean {
+    return provinciasHabilitadas.sumBy { p -> p.poblacion } >= 10000000
+  }
 }
 
 class ComercioCorresponsal(val ciudades: List<Ciudad>) : Vendedor() {
   override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
     return ciudades.contains(ciudad)
   }
+  override fun esInfluyente(): Boolean {
+    return (ciudades.size >= 5) or (this.provincias().size >= 3)
+  }
+  fun provincias() = ciudades.map { c -> c.provincia }.toSet()
+}
+//en el centro el arg ciudad se para por parametro, pero vendedores no
+class CentroDistribucion(val ciudad: Ciudad) {
+  val vendedores = mutableListOf<Vendedor>()
+  fun agregarVendedor(vendedor: Vendedor) {
+    if (vendedores.contains(vendedor)) {
+      throw Exception("El vendedor ya está registrado")
+    } else {
+      vendedores.add(vendedor)
+    }
+  }
```

Comment:
El `else` está de más, si entra en la rama de la excepción, el `add` nunca se va a ejecutar.

```suggestion
  fun agregarVendedor(vendedor: Vendedor) {
    if (vendedores.contains(vendedor)) {
      throw Exception("El vendedor ya está registrado")
    } 
    vendedores.add(vendedor)
  }
```